x-pack/plugins/cases/public/components/create/form_context.tsx

```diff
@@ -213,27 +95,27 @@ export const FormContext: React.FC<Props> = ({
     ]
   );
 
-  const { form } = useForm<FormProps>({
-    defaultValue: { ...initialCaseValue, ...initialValue },
+  const { form } = useForm({
+    defaultValue: {
+      /**
+       * This is needed to initiate the connector
+       * with the one set in the configuration
+       * when creating a case.
+       */
+      ...getInitialCaseValue({
+        owner: currentConfiguration.owner,
```

Comment:
I was testing your PR and I cannot create cases because `owner is required and must be a string`.

<img width="350" alt="Screenshot 2024-06-25 at 15 18 33" src="https://github.com/elastic/kibana/assets/1533137/9ec831db-96d4-4a04-84ba-1c7061409472">

I think the reason is this line @cnasikas .

1. `currentConfiguration` comes [from here](https://github.com/elastic/kibana/pull/185880/files#diff-2bacf9ee6f70ea2d4cea671efafd443176274c6be1c586ccca8c387d297ef6e8R154)
2.  See the implementation of `getConfigurationByOwner` in [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/public/containers/configure/utils.ts#L32).

An owner exists but if `!configurations`(what I tested) we return the `initialConfiguration` where `owner: ''`.